src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -397,6 +397,22 @@ public static ImmutableArray<TResult> SelectAsArray<TSource, TResult>(this IRead
             return ImmutableCollectionsMarshal.AsImmutableArray(builder);
         }
 
+        public static ImmutableArray<TResult> SelectAsArray<TSource, TResult, TArg>(this IReadOnlyCollection<TSource>? source, Func<TSource, TArg, TResult> selector, TArg arg)
+        {
+            if (source == null)
+                return ImmutableArray<TResult>.Empty;
+
+            var builder = new TResult[source.Count];
+            var index = 0;
+            foreach (var item in source)
```

Comment:
>foreach [](http://example.com/codeflow?start=12&length=7)

Arg, and now I noticed that the code I copied this from should probably be using a for loop instead of a foreach to avoid the enumerator allocation. Definitely willing to make that change too, but will hold off until a compiler person indicates their preference.

Reply:
Ignore this comment, this takes in an IReadOnlyCollection, not an IReadOnlyList like my mind mapped it to. The foreach is just fine.